lib/analyze-component.js

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+const { walk } = require('./walk');
+
+const componentCache = new WeakMap();
+
+function getModuleScopedFunctions(root) {
+    return new Set(
+        root.body
+            .map((node) => {
+                if (node.type === 'FunctionDeclaration') {
+                    return (node.id && node.id.name) || '';
+                } else if (
+                    node.type === 'VariableDeclaration' &&
+                    node.declarations.length === 1 &&
+                    node.declarations[0].id.type === 'Identifier' &&
+                    ((node.declarations[0].init &&
+                        node.declarations[0].init.type === 'ArrowFunctionExpression') ||
+                        (node.declarations[0].init &&
+                            node.declarations[0].init.type === 'FunctionExpression'))
+                ) {
+                    return node.declarations[0].id.name;
+                }
+                return '';
+            })
+            .filter(Boolean),
+    );
+}
+
+function importsLightningElementClass(root) {
+    return root.body
+        .map(
+            (topLevelNode) =>
+                topLevelNode.type === 'ImportDeclaration' &&
+                topLevelNode.source.value === 'lwc' &&
+                topLevelNode.specifiers.find(
+                    (importSpecifier) =>
+                        'imported' in importSpecifier &&
+                        importSpecifier.imported.name === 'LightningElement',
+                ) &&
+                true,
+        )
+        .includes(true);
+}
+
+function getClassDeclaration(root) {
+    // TODO: iterate over method names to look for renderedCallback, connectedCallback, etc
+    //       and use this to identify possible LWCs that don't import LightningElement directly
+
+    return (
+        root.body
+            .map((topLevelNode) => {
+                let classDeclaration;
+                if (topLevelNode.type === 'ClassDeclaration') {
+                    classDeclaration = topLevelNode;
+                } else if (
+                    topLevelNode.type === 'ExportDefaultDeclaration' &&
+                    topLevelNode.declaration.type === 'ClassDeclaration'
+                ) {
+                    classDeclaration = topLevelNode.declaration;
+                } else {
+                    return false;
+                }
+
+                return (
+                    classDeclaration.superClass &&
+                    classDeclaration.superClass.type === 'Identifier' &&
+                    classDeclaration.superClass.name === 'LightningElement' &&
+                    classDeclaration
+                );
+            })
+            .filter(Boolean)[0] || null
+    );
+}
+
+function getMethodInterdependencies(classDecl) {
```

Comment:
When this was originally written in TypeScript, what this function did was a bit more apparent. A description would really help to make it clear what this method does & returns. Something like: This function analyzes a class declaration and determines which methods are invoked by other methods.